File: server/src/api/dto/post.go
package dto

import (
	"time"
)

type PostCreate struct {
	Text string `json:"text"`
}

type Post struct {
	ID        int                 `json:"id"`
	Text      string              `json:"text"`
	AuthodID  int                 `json:"author_id"`
	CreatedAt time.Time           `json:"created_at"`
	UpdatedAt time.Time           `json:"updated_at"`
	Photos    []*PhotoAttachement `json:"photos"`
}

File: client-angular/src/models/post.ts
import { PhotoAttachement } from "./photoAttachement";

export class LikePost {
    post_id = 0
	user_id = 0
}

export class Post {
    id = 0;
    text = "";
    author_id = 0;
    chat_id = 0;
    created_at = new Date();
    updated_at = new Date();
	photos: PhotoAttachement[] = []
    likes: LikePost[] = []
    has_current_user_like = false;

    current_likes = 0
}
